cube_holder.scad

```diff
@@ -17,29 +17,39 @@ cube_holder();
 // Begin modules that can be used to assemble this piece in solver_3.scad
 
 module above_cube_holder() {
-  up(thickness/2 + servo_gear_thickness) above_frame() children();
+  up(servo_gear_thickness) above_frame() children();
 }
 
 module cube_holder() {
+  up(servo_gear_thickness) onto_point()
+    piece_sleeve();
+  holder_gear();
+}
+
+module piece_sleeve() {
   piece_size = cube_size / 3 + margin;
   holder_open_size = piece_size + thickness;
   holder_closed_size = piece_size + thickness + embedded_thickness;
-  difference() {
-    up(servo_gear_thickness - thickness/2)
-      onto_point(holder_closed_size)
-      difference() {
+  translate([-piece_size/2,piece_size/2,piece_size/2]) {
+    difference() {
+      translate([thickness/2, (embedded_thickness-thickness)/2, (embedded_thickness-thickness)/2])
```

Comment:
Two things going on: one, (embedded_thickness-thickness)/2 should be extracted into a var. Two, a lot of "..../2, .../2"s etc indicates to me that this piece would be clearer if it weren't designed centered.

You could design it uncentered, then, knowing the overall size, center it in one place right before calling onto_point with that size, and I think it'd be cleaner.

Reply:
Done!  Many less /2s